Scuti/Scripts/UI/Store/Offer Details/UIPanningAndPinchImage.cs

```diff
@@ -0,0 +1,334 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine.EventSystems;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+namespace Scuti.UI
+{
+    public class UIPanningAndPinchImage : MonoBehaviour, IDragHandler, IBeginDragHandler, IEndDragHandler,
+ /*IDragHandler,*/ IScrollHandler
+    {
+
+        [Header("Sub Widget")]
+        [SerializeField] private UIDisplayImageZoom displayLargeImage;
+
+        private Vector3 initialScale;
+        [SerializeField] private ScrollRect scrollRect;
+        [SerializeField] private float zoomSpeed = 0.1f;
+        [SerializeField] private float maxZoomMobile = 10f;
+        [SerializeField] private float maxZoomDesktop = 5f;
+        [SerializeField] private bool isScale;
+        [SerializeField] private bool isEnterToImage;
+        [SerializeField] private bool isDragDetected;
+
+        private RectTransform rectScroll;
+        PointerEventData cachedEventData;
+        private int counterTouch;
+        private Vector2 lastMousePosition;
+        private Vector2 averagePointBetweenTouch;
+
+        [Header("Testing")]
+        [SerializeField]
+        bool isTestingMobile;
+
+
+        // Start is called before the first frame update
+        void Awake()
+        {
+            initialScale = transform.localScale;
+            scrollRect = GetComponentInParent<ScrollRect>();
+            rectScroll = scrollRect.GetComponent<RectTransform>();
```

Comment:
@Jpablogs8 These should be serialized properties and then be assigned in the editor rather than calling GetComponent